src/live_restore/live_restore_fs.c

```diff
@@ -322,6 +329,9 @@ __live_restore_fs_directory_list_worker(WT_FILE_SYSTEM *fs, WT_SESSION *wt_sessi
             }
     }
 
+    if (lr_fs->finished)
```

Comment:
This is still susceptible to a race where we unlink the source dir after this check but while we loop the contents of the source directory. Same comment about a FIXME and dealing with it in the next ticket.

Reply:
a1c41de00